metricbeat/module/apache/status/status.go

```diff
@@ -32,22 +32,19 @@ var (
 )
 
 func init() {
-	if err := helper.Registry.AddMetricSeter("apache", "status", New); err != nil {
+	if err := mb.Registry.AddMetricSet("apache", "status", New); err != nil {
 		panic(err)
 	}
 }
 
-// New creates new instance of MetricSeter
-func New() helper.MetricSeter {
-	return &MetricSeter{}
-}
-
-type MetricSeter struct {
+// MetricSet for fetching Apache HTTPD server status.
+type MetricSet struct {
+	mb.BaseMetricSet
```

Comment:
Yes, the BaseMetricSet type can be embedded into another struct to satisfy the MetricSet interface requirements, leaving only the Fetch() method to be implemented to have a complete MetricSet implementation.

Without embedding it, then it would need to have a `Name()` and `Module()` method implemented to be a `MetricSet`.